vllm/sampling_params.py

```diff
@@ -316,6 +317,26 @@ def clone(self) -> "SamplingParams":
         }
         return copy.deepcopy(self, memo=logit_processor_refs)
 
+    def get_logits_processors(self):
+        logits_processors = []
+
+        for lp in self.logits_processors or []:
+            if isinstance(lp, LogitsProcessorFactory):
+                logits_processors.append(lp.get_processor())
+            else:
+                logits_processors.append(lp)
+        return logits_processors
```

Comment:
Could be more concise, similar for below
```suggestion
        return [] if not self.logits_processors else [
           lp.get_processor() if isinstance(lp, LogitsProcessorFactory) else lp
           for lp in self.logits_processors
        ]
```